### PR TITLE
Feature/openads2 add position

### PR DESCRIPTION
**Pull request by victuxbb**
This PR includes:
* Add position use case
* Raise events from Domain Position
* Observer for Position creation to call appnexus side effects 
* ProxyPositionHandler to solve consumers problem in appnexus connector

### CODE REVIEW

File: src/openads/infrastructure/position/proxyHandlerFactory.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import {AD_AVAILABLE} from '../connector/appnexus/event/events'
+import AppNexusErrorException from '../connector/appnexus/AppNexusErrorException'
+
+const PROXY_PROPERTY = 'ad'
+const TIMEOUT = 60
```
Comment (alextremp) [inline src/openads/infrastructure/position/proxyHandlerFactory.js]: It seems a very short time to wait...

File: src/openads/infrastructure/position/proxyHandlerFactory.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        cause: target[name],
+        message: APPNEXUS_ERROR,
+        status: target[name].status
+      }))
+      break
+  }
+}
+let timeout = () => new Promise((resolve, reject) => {
+  let wait = setTimeout(() => {
+    clearTimeout(wait)
```
Comment (alextremp) [inline src/openads/infrastructure/position/proxyHandlerFactory.js]: There's no need to clear a timeout, as you can't use it 2 times, only a setInterval

Comment (victuxbb) [reply src/openads/infrastructure/position/proxyHandlerFactory.js]: https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-cleartimeout
Seems a good idea to clean it form global list of active timers...

Comment (alextremp) [reply src/openads/infrastructure/position/proxyHandlerFactory.js]: 👍

File: src/openads/infrastructure/position/positionCreatedObserver.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import {AD_AVAILABLE, AD_BAD_REQUEST, AD_ERROR, AD_NO_BID, AD_REQUEST_FAILURE} from '../connector/appnexus/event/events'
+
+const positionCreatedObserverFactory = appnexusConnector => appNexusConsumersRepository => ({payload, dispatcher}) =>
```
Comment (rafamarquezv) [inline src/openads/infrastructure/position/positionCreatedObserver.js]: Should not this observer be registered in the Container?

Comment (victuxbb) [reply src/openads/infrastructure/position/positionCreatedObserver.js]: You are completely right, I'm missing the build method and the registration to the Domain Event Bus!

File: src/openads/domain/service/observerErrorThrown.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const OBSERVER_ERROR_THROWN = 'OBSERVER_ERROR_THROWN'
+export const observerErrorThrown = payload => ({type: OBSERVER_ERROR_THROWN, payload})
```
[the remark this answers is not in the document]
Comment (victuxbb) [reply src/openads/domain/service/observerErrorThrown.js]: Fixed!

File: src/openads/infrastructure/connector/appnexus/AppNexusErrorException.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export default class AppNexusErrorException extends Error {
```
Comment (alextremp) [inline src/openads/infrastructure/connector/appnexus/AppNexusErrorException.js]: how an OpenAds client would know that the recevied Error is an AppNexusErrorException?
Shouldn't it be exported?

File: src/openads/infrastructure/position/InMemoryPositionRepository.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+  /**
+   * Store the position in memory
+   * @param {Position} position
+   * @returns {Promise<Position>}
+   */
+  save ({position}) {
+    return this.find({id: position.id})
+      .then(position => {
+        if (position !== null) throw new Error('PositionAlreadyExists')
```
Comment (alextremp) [inline src/openads/infrastructure/position/InMemoryPositionRepository.js]: Maybe domain exception here?

File: src/openads/infrastructure/connector/appnexus/AppNexusErrorException.js (hunk 5 above)
Comment (victuxbb) [reply src/openads/infrastructure/connector/appnexus/AppNexusErrorException.js]: I think the best solution is to export types that are necessary for the client, so I will modify the index and expose all necessary things :+1:

**Comment by victuxbb** [comment]
All tests green!

**Pull request merged**